### PR TITLE
[Bazel/MSVC] Cleanup flags and config_settings for MSVC

### PR DESCRIPTION
**Pull request by rongjiecomputer**

### CODE REVIEW

**Comment by rongjiecomputer** [comment]
PTAL @mrry

**Comment by meteorcloudy** [comment]
@rongjiecomputer Thank you so much for doing those clean up work!
Could you also remove all usage of `//tensorflow:windows_msvc` config_settting? Because that's a legacy config_setting which is now identical to `//tensorflow:windows`.

File: third_party/snappy.BUILD [hunk 1]
```diff
@@ -18,17 +18,10 @@ cc_library(
...
-            "/DHAVE_CONFIG_H",
-            "/EHsc",
-        ],
-        "@org_tensorflow//tensorflow:windows_msvc": [
-            "/DHAVE_CONFIG_H",
-            "/EHsc",
-        ],
+    copts = ["-DHAVE_CONFIG_H"] + select({
+        "@org_tensorflow//tensorflow:windows": [],
+        "@org_tensorflow//tensorflow:windows_msvc": [],
```
Comment (meteorcloudy) [inline third_party/snappy.BUILD]: We can remove this

**Comment by rongjiecomputer** [comment]
> Could you also remove all usage of //tensorflow:windows_msvc config_settting? Because that's a legacy config_setting which is now identical to //tensorflow:windows.

Done. I have removed all references to `:windows_msvc` I can find with Github search. Let's see if this break CI build.

**Comment by rongjiecomputer** [comment]
@meteorcloudy After our last conversation, one more `windows_msvc` slipped into the codebase https://github.com/tensorflow/tensorflow/commit/158cd6220231fcf758a45c2dcd40d93cd0aec9e0#diff-15609f78c0cd35bfdc31f9d8bd9b1e91R2171.

I have rebased and removed the `windows_msvc`.

Please help to spread awareness to internal team to not use `windows_msvc` anymore.

**Comment by martinwicke** [comment]
@meteorcloudy maybe we can use a METADATA warning to prevent new occurrences.

**Comment by meteorcloudy** [comment]
Thanks @martinwicke , will do it!

**Review by meteorcloudy — DISMISSED** [review]

**Comment by tensorflowbutler** [comment]
Nagging Assignee @martinwicke: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by rongjiecomputer** [comment]
Looks like my change is pushed from internal already. Closing this now. 😄

**Pull request closed without merging**